framerail/src/routes/[slug]/[...extra]/VotePane.svelte

```diff
@@ -0,0 +1,152 @@
+      })
+    } else {
+      voteMap = new Map()
+      res.forEach((vote) => {
+        voteMap.set(vote.user_id, vote)
+      })
+    }
+  }
+
+  async function castVote(value?: number) {
+    let fdata = new FormData()
+    fdata.set("site-id", $page.data.site.site_id)
+    fdata.set("page-id", $page.data.page.page_id)
+    fdata.set("action", "set")
+    fdata.set("value", value ?? 0)
+    let res = await fetch(`/${$page.data.page.slug}/vote-cast`, {
+      method: "POST",
+      body: fdata
+    }).then((res) => res.json())
+    if (res?.message) {
+      showErrorPopup.set({
+        state: true,
+        message: res.message,
+        data: res.data
+      })
+    }
+  }
+
+  async function cancelVote() {
+    let fdata = new FormData()
+    fdata.set("site-id", $page.data.site.site_id)
+    fdata.set("page-id", $page.data.page.page_id)
+    fdata.set("action", "remove")
+    let res = await fetch(`/${$page.data.page.slug}/vote-cancel`, {
+      method: "POST",
+      body: fdata
+    }).then((res) => res.json())
+    if (res?.message) {
+      showErrorPopup.set({
+        state: true,
+        message: res.message,
+        data: res.data
+      })
+    }
+  }
+
+  onMount(async () => {
+    let fdata = new FormData()
+    fdata.set("site-id", $page.data.site.site_id)
+    fdata.set("page-id", $page.data.page.page_id)
+    let res = await fetch(`/${$page.data.page.slug}/score`, {
+      method: "POST",
+      body: fdata
+    }).then((res) => res.json())
+    if (res?.message) {
+      showErrorPopup.set({
+        state: true,
+        message: res.message,
+        data: res.data
+      })
+    } else {
+      voteRating = res.score ?? 0
+    }
+  })
+</script>
+
+<div class="vote-panel">
+  <div class="action-row vote-action">
+    <button
+      class="action-button view-vote-list clickable"
+      type="button"
+      on:click|stopPropagation={() => {
+        getVoteList().then(() => {
+          showVoteList = true
+        })
+      }}
+    >
+      {$page.data.internationalization?.["wiki-page-vote-list"]}
+    </button>
+    <div class="action-button vote-rating">
+      <span class="vote-desc"
+        >{$page.data.internationalization?.["wiki-page-vote-score"]}</span
+      >
+      <span class="vote-rating-number">{voteRating}</span>
+    </div>
+    <div class="action-button cast-vote">
+      <span class="vote-desc"
+        >{$page.data.internationalization?.["wiki-page-vote-set"]}</span
+      >
+      <button
+        class="vote-subbutton clickable"
+        type="button"
+        on:click|stopPropagation={() => castVote(1)}
+      >
+        +1
+      </button>
+      <button
+        class="vote-subbutton clickable"
+        type="button"
+        on:click|stopPropagation={() => castVote(0)}
+      >
+        0
+      </button>
+      <button
+        class="vote-subbutton clickable"
+        type="button"
+        on:click|stopPropagation={() => castVote(-1)}
+      >
+        -1
+      </button>
```

Comment:
Note: This is for the standard vote scheme but we will need to fit page vote settings into here at some point (e.g. disabled voting and 5-star).

Reply:
Created [WJ-1294 - UI for alternate voting mechanisms](https://scuttle.atlassian.net/browse/WJ-1294)